### PR TITLE
MIGENG-358: Numa Pinnings Flag rule and tests

### PR DESCRIPTION
**Pull request by m-brophy**
https://issues.redhat.com/browse/MIGENG-358

This PR requires work in xavier-integration https://github.com/project-xavier/xavier-integration/pull/84 to work end-to-end

### CODE REVIEW

**Review by carlosthe19916 — COMMENTED** [review]
@m-brophy The code and the tests look good to me.

**Review by mrizzi — DISMISSED** [review]
thanks @m-brophy for this new rule :+1:

File: src/main/java/org/jboss/xavier/analytics/functions/HelperFunctions.java [hunk 1]
```diff
@@ -85,7 +85,8 @@ boolean isConvertible()
     public enum FlagUnsuitabilityForOSPTarget{
         RDM_DISK(WorkloadInventoryReportModel.RDM_DISK_FLAG_NAME, true),
         TOO_MANY_NICS(WorkloadInventoryReportModel.MORE_THAN_4_NICS_FLAG_NAME, true),
-        SHARED_DISK(WorkloadInventoryReportModel.SHARED_DISK_FLAG_NAME,true);
+        SHARED_DISK(WorkloadInventoryReportModel.SHARED_DISK_FLAG_NAME,true),
+        NUMA_PINNINGS(WorkloadInventoryReportModel.NUMA_PINNINGS_FLAG_NAME, false);
```
Comment (mrizzi) [inline src/main/java/org/jboss/xavier/analytics/functions/HelperFunctions.java]: Following https://issues.redhat.com/browse/MIGENG-358?focusedCommentId=14000961&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-14000961, please change this to become `CPU_AFFINITY`

**Review by mrizzi — CHANGES_REQUESTED** [review]
@m-brophy there has been a change in requirement https://issues.redhat.com/browse/MIGENG-358?focusedCommentId=14000961&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-14000961 and hence this PR has to change consistently.

**Comment by mmmmmmpc** [comment]
Could we please change the name of this flag to "CPU affinity"

**Comment by carlosthe19916** [comment]
@m-brophy Thanks for all the changes... I've tested your last changes in CI and it worked fine. I've also reported a couple of screenshots from the test in this comment: https://github.com/project-xavier/xavier-integration/pull/84#pullrequestreview-376762526

**Review by carlosthe19916 — APPROVED** [review]

**Pull request merged**